kubernetes/fabric8/src/main/java/org/arquillian/cube/kubernetes/fabric8/impl/visitor/SecretsAndServiceAccountVisitor.java

```diff
@@ -0,0 +1,126 @@
+        String serviceAccount = null;
+        Set<Secret> secrets = new LinkedHashSet<>();
+
+        if (element instanceof PodBuilder) {
+            PodBuilder builder = (PodBuilder) element;
+            serviceAccount = builder.getSpec().getServiceAccountName();
+            secrets.addAll(generateSecrets(builder.getMetadata()));
+        } else if (element instanceof PodTemplateSpecBuilder) {
+            PodTemplateSpecBuilder builder = (PodTemplateSpecBuilder) element;
+            serviceAccount = builder.getSpec().getServiceAccountName();
+            secrets.addAll(generateSecrets(builder.getMetadata()));
+        }
+
+    }
+
+
+
+    private void createServiceAccount(String serviceAccount, Set<Secret> secrets) {
+
+        KubernetesClient client = this.client.get();
+        Configuration configuration = this.configuration.get();
+
+        List<ObjectReference> refs = new ArrayList<>();
+        for (Secret secret : secrets) {
+            refs.add(
+                    new ObjectReferenceBuilder()
+                            .withNamespace(configuration.getNamespace())
+                            .withName(secret.getMetadata().getName())
+                            .build()
+            );
+        }
+
+
+        if (client.serviceAccounts().inNamespace(configuration.getNamespace()).withName(serviceAccount).get() == null) {
+            client.serviceAccounts().inNamespace(configuration.getNamespace()).createNew()
+                    .withNewMetadata()
+                    .withName(serviceAccount)
+                    .endMetadata()
+                    .withSecrets(refs)
+                    .done();
+        } else {
+            client.serviceAccounts().inNamespace(configuration.getNamespace()).withName(serviceAccount).edit()
+                    .withSecrets(refs)
+                    .done();
+        }
+    }
+
+    private Set<Secret> generateSecrets(ObjectMeta meta) {
+        KubernetesClient client = this.client.get();
+        Configuration configuration = this.configuration.get();
+
+        Set<Secret> secrets = new HashSet<>();
+        Map<String, String> annotations = meta.getAnnotations();
+        if (annotations != null && !annotations.isEmpty()) {
+            for (Map.Entry<String, String> entry : annotations.entrySet()) {
+                String key = entry.getKey();
+                String value = entry.getValue();
+                if (SecretKeys.isSecretKey(key)) {
+                    SecretKeys keyType = SecretKeys.fromValue(key);
+                    for (String name : Secrets.getNames(value)) {
+                        Map<String, String> data = new HashMap<>();
+
+                        Secret secret = null;
+                        try {
+                            secret = client.secrets().inNamespace(configuration.getNamespace()).withName(name).get();
+                        } catch (Exception e) {
+                            // ignore - probably doesn't exist
```

Comment:
I think that the whole try/catch should go. No checked execeptions are thrown from this piece of code.